src/main/java/net/minecraftforge/event/entity/player/PlayerSpawnPhantomsEvent.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.event.entity.player;
+
+import net.minecraft.world.entity.player.Player;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.eventbus.api.Cancelable;
+import net.minecraftforge.eventbus.api.Event;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Fired from {@link net.minecraft.world.level.levelgen.PhantomSpawner} per player when it attempts to spawn phantoms.<br>
+ * <br>
+ * This event fires before any checks (but <i>after</i> check if player is a spectator and
+ * <i>after</i> global checks, such as dimension being dark enough or doInsomnia is true)
+ * and generally allow you to change properties of <i>potential</i> phantom spawns through
+ * {@link PlayerSpawnPhantomsEvent#setPhantomsToSpawn}.<br>
+ * <br>
+ * Behavior of {@link net.minecraft.world.level.levelgen.PhantomSpawner} is determined by {@link PlayerSpawnPhantomsEvent#getResult()}.<br>
+ * See {@link PlayerSpawnPhantomsEvent#setResult} for documentation.<br>
+ * <br>
+ * This event is not {@link Cancelable}.<br>
+ * <br>
+ * This event is fired on the {@link MinecraftForge#EVENT_BUS}.<br>
+ */
+@Event.HasResult
```

Comment:
```
/**
 * This event is fired from {@link PhantomSpawner#tick}, once per player, when phantoms would attempt to be spawned.<br>
 * This event is not fired for spectating players.
 * <p>
 * This event is fired before any per-player checks, but after all global checks.<br>
 * The behavior of {@link PhantomSpawner} is determined by the result of this event.<br>
 * See {@link #setResult} for documentation.<br>
 * <p>
 * This event is fired on the {@link MinecraftForge#EVENT_BUS}.<br>
 * @see {@link #setResult} for the effects of each result.
 */
 ```
 
 The file should also import `net.minecraft.world.level.levelgen.PhantomSpawner` for the javadocs to resolve correctly.